changes/2024-TBD_cache-across-hierarchy-keyrings/change.md

```diff
@@ -0,0 +1,613 @@
+the entire appended string
+(a part of which is the hash of the branch-key-id)
+to get the final cache identifier.
+We'll call these internal and external hashes.
+
+### Issue 1: How to update the cache identifier for the Hierarchy Keyring to allow a shared cache?
+
+Note that according to the current
+[Hierarchy Keyring Cache Identifier formula](#hierarchy-keyring-cache-identifier-formula),
+if two KMS Relationships Hierarchy Keyrings are
+called with two different Key Stores but the same branch-key-id,
+there will be a collision (the branch-key-version will also
+have to co-incide in case of Decryption Branch Key Materials).
+
+#### Option 1 (Recommended): Append the keystore-id to the [Hierarchy Keyring cache identifier formula](#hierarchy-keyring-cache-identifier-formula).
+
+By appending the [`keystore-id`](../../framework/branch-key-store.md#keystore-id) to the
+Hierarchy Keyring cache identifier,
+we can ensure that there are no collisions between two identifiers
+from different local Key Stores.
+
+Key Store IDs are UUIDs that are unlikely to collide by default.
+
+By adding Key Store ID to the Hierarchy Keyring cache identifier,
+the customer can control cache access.
+For instance, they can choose if different
+KMS Relationships CAN hit the same identifier or not.
+They can also choose if different logical key stores
+or physical key stores can collide or not.
+But by default, because each Key Store object will have a unique UUID,
+there will be no collision.
+More context on this can be found in the
+[Appendix section A1](#a1-relating-cache-entries-branch-keys-to-kms-relationships).
+
+#### Option 3: Don't change anything.
+
+As mentioned before, in-case of a shared cache,
+there will be collisions while fetching materials
+from multiple Hierarchy Keyrings.
+
+### Issue 2: Should we update the cache identifier for the Hierarchy Keyring BranchKeyMaterials to include branch-key-digest instead of branch-key-id?
+
+Note that according to the current
+[Hierarchy Keyring Cache Identifier formula](#hierarchy-keyring-cache-identifier-formula),
+the [Hierarchy Keyring Branch Key Materials (Decryption) cache identifier](https://github.com/awslabs/aws-encryption-sdk-specification/tree/ffb2b0cc6a956b2cec3a33be3c3672605b6907fb/framework/aws-kms/aws-kms-hierarchical-keyring.md#decryption-materials)
+has branch-key-id appended in the cache identifier formula and the
+[Branch Key Materials (Encryption) cache identifier](https://github.com/awslabs/aws-encryption-sdk-specification/tree/ffb2b0cc6a956b2cec3a33be3c3672605b6907fb/framework/aws-kms/aws-kms-hierarchical-keyring.md#encryption-materials)
+has branch-key-digest=SHA512(branch-key-id). Here we're talking
+about the internal hash.
+
+#### Option 1 (Recommended): Update the Decryption cache identifier to include branch-key-digest=SHA512(branch-key-id) instead of the branch-key-id
+
+This will make both the Hierarchy Keyring Encryption and
+Decryption cache identifiers more coherent.
```

Comment:
nit: Do you mean consistent? I think the formulas are coherent but not consistent.